ocaml/xapi/import.ml

```diff
@@ -547,36 +549,92 @@ module VDI : HandlerTools = struct
 				Found_no_iso
 		end else begin
 			match config.import_type with
-			| Metadata_import _ -> begin
+			| Metadata_import { vdi_map } -> begin
 				let mapto =
 					if List.mem_assoc Constants.storage_migrate_vdi_map_key vdi_record.API.vDI_other_config
 					then Some (Ref.of_string (List.assoc Constants.storage_migrate_vdi_map_key vdi_record.API.vDI_other_config))
 					else None in
-				let find_by_sr_and_location () =
-					let sr = lookup vdi_record.API.vDI_SR state.table in
-					List.filter (fun (_, vdir) ->
-						vdir.API.vDI_location = vdi_record.API.vDI_location && vdir.API.vDI_SR = sr)
-						(Client.VDI.get_all_records rpc session_id) |> choose_one |> Opt.map fst in
-				match (
-					if exists vdi_record.API.vDI_SR state.table
-					then match find_by_sr_and_location () with
+				let vdi_records = Client.VDI.get_all_records rpc session_id in
+				let find_by_sr_and_location sr location =
+					vdi_records
+					|> List.filter (fun (_, vdir) -> vdir.API.vDI_location = location && vdir.API.vDI_SR = sr)
```

Comment:
I know we used to define `|>` ourselves, but I see that `|>` was used in this file even before the changes made by this pull request, so it should be fine.